inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -376,6 +376,49 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
     }
 }
 
+void FrontEnd::processTrivialCases(const Model& model) {
+    std::map<ie::DataPtr, DataVector> ieToVpuDataVector;
+    std::for_each(model->datas().begin(), model->datas().end(), [&ieToVpuDataVector](const Data& data) {
+        ieToVpuDataVector[data->origData()].push_back(data);
+    });
+
+    std::vector<DataVector> trivialCases;
+    std::for_each(ieToVpuDataVector.begin(), ieToVpuDataVector.end(),
+            [&trivialCases](const std::pair<ie::DataPtr, DataVector>& ieToDataVector) {
+        if (ieToDataVector.second.size() > 1) {
+            trivialCases.push_back(ieToDataVector.second);
+        }
+    });
+
+    for (const auto& trivialCase : trivialCases) {
+        Data trivialInput, trivialOutput;
+        for (const auto& data : trivialCase) {
+            if (data->usage() == DataUsage::Input || data->usage() == DataUsage::Intermediate || data->usage() == DataUsage::Const) {
+                if (!data->isConsumed()) {
+                    trivialInput = data;
+                }
+            }
+            if (data->usage() == DataUsage::Output || data->usage() == DataUsage::Intermediate) {
+                if (data->producer() == nullptr && data->childDataToShapeEdges().size() == 0) {
+                    trivialOutput = data;
+                }
+            }
+        }
```

Comment:
So, `trivialInput` and `trivialOutput` contains just the last *trivial* data object from collection?
BTW, I don't think *trivial* is a good term for data objects. For network in general it is... acceptable at least

Reply:
Changed to `unconnectedInput` and `unconnectedOutput`, actually they should be the only ones, added a check on that.

Reply:
Don't see any changes. Are you sure you've committed your latest update?

Reply:
Ops, my fault. sorry. Pushed.